paddle/fluid/operators/prelu_op.cc

```diff
@@ -25,15 +22,36 @@ class PReluOp : public framework::OperatorWithKernel {
           const framework::AttributeMap &attrs)
       : OperatorWithKernel(type, inputs, outputs, attrs) {}
 
-  void InferShape(framework::InferShapeContext *ctx) const override {
+  void InferShape(framework:: InferShapeContext *ctx) const override {
+
+    std::string mode = ctx->Attrs().Get<std::string>("mode");
+
+    auto dim = ctx->GetInputDim("X");
     PADDLE_ENFORCE(ctx->HasInput("X"), "Input(X) should not be null");
     PADDLE_ENFORCE(ctx->HasInput("Alpha"), "Input(Alpha) should not be null");
-    PADDLE_ENFORCE(product(ctx->GetInputDim("Alpha")) == 1,
-                   "Size of weight Alpha must be one.");
+    
     PADDLE_ENFORCE(ctx->HasOutput("Out"), "Output(Out) should not be null");
+    if (mode == "all")
+      PADDLE_ENFORCE(product(ctx->GetInputDim("Alpha")) == 1,
```

Comment:
product(ctx->GetInputDim("Alpha")) ->  alpha_dim.numel()

Reply:
The same issue as before